sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/SecureConnectionProvider.scala

```diff
@@ -26,6 +26,12 @@ import org.apache.spark.internal.Logging
 import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions
 import org.apache.spark.util.SecurityUtils
 
+/**
+ * Some of the secure connection providers modify global JVM security configuration.
+ * In order to avoid race the modification must be synchronized with this.
+ */
+private[connection] object SecurityConfigurationLock
```

Comment:
When other parts of the code is also modifying the global JVM security config then it must be moved into a global place (and must be used).